File: LocationListResource/src/main/java/com/djh/location/list/core/service/SummaryService.java
package com.djh.location.list.core.service;

import com.djh.location.list.core.domain.SummaryItem;
import com.djh.location.list.core.domain.SummaryItems;

/**
 * @author David Hancock
 */
public interface SummaryService {

    public SummaryItems findAllSummaryItems();

    public void addSummaryItem(SummaryItem summaryItem);

    public void deleteSummaryItem(String summaryItemId);

    public void updateSummaryItem(String summaryItemId, SummaryItem summaryItem);

}

File: LocationListWeb/app/resources/js/summary/summary-service.js
summaryModule.service('summaryService', ['$rootScope', '$http', '$location', function ($rootScope, $http, $location) {

    const PORT = '8089';

    const HOST = $location.host();

    const URL = 'http://' + HOST + ':' + PORT + '/summary/';

    this.editSummaryItem = function (summaryItem) {
        return $http({method: 'PUT', url: URL + summaryItem.id, data: summaryItem}).
            success(function (data) {
                return data;
            }).error(function (data, status, headers, config) {
                $rootScope.$broadcast('error:http', status);
            });
    };

    this.addSummaryItem = function (summaryItem) {
        return $http({method: 'POST', url: URL, data: summaryItem}).
            success(function (data) {
                return data;
            }).error(function (data, status, headers, config) {
                $rootScope.$broadcast('error:http', status);
            });
    };

    this.deleteSummaryItem = function (summaryItemId) {
        return $http({method: 'DELETE', url: URL + summaryItemId}).
            success(function (data) {
                return data;
            }).error(function (data, status, headers, config) {
                $rootScope.$broadcast('error:http', status);
            });
    };

    this.getSummaryItems = function () {
        return $http({method: 'GET', url: URL}).
            success(function (data) {
                return data;
            }).error(function (data, status, headers, config) {
                $rootScope.$broadcast('error:http', status);
            });
    };

}]);